android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/home/HomeScreen.kt

```diff
@@ -2,21 +2,43 @@ package co.nimblehq.ic.kmm.suv.android.ui.screens.home
 
 import androidx.compose.foundation.background
 import androidx.compose.foundation.layout.*
-import androidx.compose.foundation.shape.CircleShape
-import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.draw.clip
 import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
-import androidx.compose.ui.unit.dp
-import co.nimblehq.ic.kmm.suv.android.R
-import co.nimblehq.ic.kmm.suv.android.ui.theme.Typography
-import coil.compose.AsyncImage
+import co.nimblehq.ic.kmm.suv.android.ui.components.ErrorAlertDialog
+import org.koin.androidx.compose.getViewModel
 
 @Composable
-fun HomeScreen() {
+fun HomeScreen(viewModel: HomeViewModel = getViewModel()) {
+    val currentDate by viewModel.currentDate.collectAsState()
+    val avatarUrlString by viewModel.avatarUrlString.collectAsState()
+    val isLoading by viewModel.isLoading.collectAsState()
+    val errorMessage by viewModel.errorMessage.collectAsState()
+
+    LaunchedEffect(Unit) {
+        viewModel.loadProfile()
+    }
+
+    errorMessage?.let {
+        ErrorAlertDialog(
+            message = it,
+            onButtonClick = { viewModel.dismissError() }
+        )
+    }
```

Comment:
To handle the error, I'd suggest using `SharedFlow.collect` to observe the error instead

```
    LaunchedEffect(Unit) {
        viewModel.errorMessage.collect { errorMessage ->
// Handle to show dialog
        }
    }
```

Reply:
As our discussion, we decided to keep the current approach.